src/core/storage/listener.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * The storage listener is a listener on changes to the SDK global storage
+ */
+export default class StorageListener {
+  /**
+   * @param {string} eventType The type of event to listen for e.g. 'update'
+   * @param {string} storageKey The key to listen for e.g. 'Pagination'
+   * @param {Function} callback The callback to call when the event is emitted on the storage key
+   */
+  constructor (eventType, storageKey, callback) {
+    this.eventType = eventType;
+    this.storageKey = storageKey;
+    this.callback = callback || function () {};
```

Comment:
I like this model! maybe we shouldn't default callback to something? if somebody didn't pass a callback to a listener that's probably a bug right? maybe the validation that is done in storage.addListener() should just be done here?